### ISSUE TITLE
RuntimeSDK: Improve Runtime Hook Swagger documentation

### ISSUE THREAD

**Issue by sbueringer** [kubernetes-sigs/cluster-api#6505]
**User Story**

As a user I would like to be able to get sufficient self-explanatory information to implement RuntimeHooks via the generated Swagger documentation/specification.

**Detailed Description**

This is a follow-up to #6455 (specifically: https://github.com/kubernetes-sigs/cluster-api/pull/6455#discussion_r870343550 & https://github.com/kubernetes-sigs/cluster-api/pull/6455#discussion_r870345945). Goal of this issue is to extend the godoc comments of the request/response types and the description of the Hook (specified when registering the hook).

We should document things like:
* when is the hook triggered
* additional comments about how responses should be calculated, which fields are mandatory, ...
* ...

Note:
* This definitely applies to the Discovery hook, potentially to the other hooks we will add as well. 

/kind cleanup
/area runtime-sdk

**Comment by sbueringer** [comment]
As the [RuntimeSDK documentation](https://release-1-2.cluster-api.sigs.k8s.io/tasks/experimental-features/runtime-sdk/index.html) has been merged now, it's probably time to revisit this issue.

I'm not sure how much we want to document per hook in the Swagger schema, given the more extensive documentation we have in the book now (afaik we can only use plain strings for documentation in the schema).

We also already did a few rounds of improving descriptions/godocs of the hook API types, which are also reflected in the Swagger scheme.

I think it makes sense to at least manually sync the YAMLs linked in the doc again with what we currently generate until #6795 is implemented.

Is there anything else that we should do apart from re-reading the schema and see if makes sense as it is rendered in the Swagger-UI?

@fabriziopandini Opinions?
